test-resources/headless-core-stub/lambdas/start/src/services/signing-service.ts

```diff
@@ -1,16 +1,72 @@
 import { GetPublicKeyCommand, KMSClient } from "@aws-sdk/client-kms";
-import { CompactEncrypt, importSPKI, KeyLike } from "jose";
+import { CompactEncrypt, importJWK, importSPKI, JSONWebKeySet, JWK, KeyLike } from "jose";
 import { HeadlessCoreStubError } from "../../../../utils/src/errors/headless-core-stub-error";
+import { formatAudience } from "../../../../utils/src/audience-formatter";
+import { Logger } from "@aws-lambda-powertools/logger";
 
 const kmsClient = new KMSClient({ region: "eu-west-2" });
+export const logger = new Logger();
 
 let cachedPublicKey: KeyLike | undefined;
 
-export const getPublicEncryptionKey = async () => {
+export const getPublicEncryptionKey = async (audience: string) => {
     if (cachedPublicKey) {
         return cachedPublicKey;
     }
 
+    if (process.env.KEY_ROTATION_FEATURE_FLAG_ENABLED === "true") {
+        await getPublicEncryptionKeyJwksUri(audience);
```

Comment:
Could we change the name of this to reflect that it updates the `cachedPublicKey` maybe something like `setCachedPublicKeyFromJwks`?